storage/aptosdb/src/event_store/mod.rs

```diff
@@ -182,6 +184,12 @@ impl EventStore {
             result.push((seq, ver, idx));
             cur_seq += 1;
         }
+        // If we didn't find any events for the given key, we should return an
+        // error to indicate that the event stream doesn't exist, at least
+        // within the specified ledger version range.
+        if !event_key_present {
+            bail!("Event key not present in the DB within ledger version range");
+        }
```

Comment:
Can we return a `Result<Option<Vec<...>>`, then we can differentiate, or at least just handle the difference between no events & no event key?

Reply:
I think that makes sense. I need to talk to the storage folks first about what is even possible first (ie the questions in the summary).